politeiawww/wsdcrdata.go

```diff
@@ -83,16 +96,100 @@ func (w *wsDcrdata) unsubFromAddr(address string) error {
 	}
 	_, err := w.client.Unsubscribe(event)
 	if err != nil {
-		return fmt.Errorf("failed to unsubscribe: %v", err)
+		return fmt.Errorf("wsDcrdata failed to unsubscribe from %v: %v",
+			event, err)
 	}
 	w.removeSub(event)
 	log.Debugf("wsDcrdata unsubscribed from addr: %v", address)
 	return nil
 }
 
-// newWSDcrdata return a new wsDcrdata context.
-func newWSDcrdata() (*wsDcrdata, error) {
-	// Init websocket client
+// subToNewBlock subscribes to dcrdata events for new blocks
+func (w *wsDcrdata) subToNewBlock() error {
+	if w.isSubscribed(newBlockSub) {
+		return errDuplicateSub
+	}
+	_, err := w.client.Subscribe(newBlockSub)
+	if err != nil {
+		return fmt.Errorf("wsDcrdata failed to subscribe to %v: %v",
+			newBlockSub, err)
+	}
+	w.addSub(newBlockSub)
+	log.Debugf("wsDcrdata subscribed to new block")
+	return nil
+}
+
+// reconnect creates a new websocket client, and subscribes to the
+// same subscriptions as the previous client.
+func (w *wsDcrdata) reconnect() {
+	done := make(chan struct{})
+
+	go func() {
+		prevSubscriptions := make(map[string]struct{}, len(w.subscriptions))
+		timeToWait := 1 * time.Minute
+
+		// Remove existing subscriptions since the client is
+		// no longer connected. These will be resubscribed to.
+		w.client.Stop()
+		for sub := range w.subscriptions {
+			prevSubscriptions[sub] = struct{}{}
+		}
+		w.removeAllSubs()
+
+		for len(prevSubscriptions) > 0 {
+			// Reconnect to dcrdata if needed
+			err := w.client.Ping()
+			if err != nil {
+				w.client.Stop()
+
+				// Existing subscriptions have gone bad
+				for sub := range w.subscriptions {
+					prevSubscriptions[sub] = struct{}{}
+				}
+				w.removeAllSubs()
+
+				// Reconnect
+				c, err := newDcrdataWSClient()
+				if err != nil {
+					log.Errorf(err.Error())
```

Comment:
You need to give your error messages context when they're being printed to the console.  Also, why are you calling the `Error()` method instead of just printing `err`?

Reply:
I was calling err.Error() because that returns a string. I've updated it now to give more context though.